megamek/src/megamek/client/bot/princess/FireControl.java

```diff
@@ -2700,10 +2777,11 @@ Mounted getPreferredAmmo(final Entity shooter,
                 return validAmmo.get(0);
             }
 
+            // Todo: delete
```

Comment:
The existing getAtmAmmo code was interfering with toHit-based ammo iteration and selection, so I've disabled it here for testing and will most likely remove it.  Same with the tests for this function.

Going forward we should be able to rely more on toHitData and weaponFireInfo for determining missile ammo to use, based on actual calculated expected damage and to-hit chances.